### PR TITLE
Global: Change the process for errors

### PR DESCRIPTION
**Pull request by muramura**
It's a meaningless process for build warnings.
I declare that I will not use the result in the preprocessor.

### CODE REVIEW

File: ArduCopter/GCS_Mavlink.cpp [hunk 1]
```diff
@@ -1397,17 +1397,19 @@ void GCS_MAVLINK_Copter::handleMessage(const mavlink_message_t &msg)
         mavlink_set_home_position_t packet;
         mavlink_msg_set_home_position_decode(&msg, &packet);
         if ((packet.latitude == 0) && (packet.longitude == 0) && (packet.altitude == 0)) {
-            if (!copter.set_home_to_current_location(true)) {
-                // silently ignored
-            }
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-result"
+            copter.set_home_to_current_location(true);
+#pragma GCC diagnostic pop
```
Comment (IamPete1) [inline ArduCopter/GCS_Mavlink.cpp:1403]:
```suggestion
            IGNORE_RETURN(copter.set_home_to_current_location(true));
```

**Review by IamPete1 — CHANGES_REQUESTED** [review]

**Comment by peterbarker** [comment]
There's *also* `UNUSED_RESULT`.... which is implemented differently... yay.  Marking for DevCall so we can choose just one to go forward with - both naming and implementation.

@lucasdemarchi was there an advantage in using the attribute rather than just chucking it into an `if()` like `IGNORE_RETURN` does?

**Comment by lucasdemarchi** [comment]
At the time that was written it was replacing the usual method of doing `(void)function();` - cast to void to ignore the result. The problem was that functions with the attribute to warn unused results was triggering the warn - see gcc bug mentioned in the commit.

As for the difference in IGNORE_RETURN vs UNUSED_RESULT, I'm not sure one is better than the other. We'd probably need to check the end result for  IGNORE_RETURN to confirm the compiler is not emitting additional code there. UNUSED_RESULT is also much older in the codebase: it was added by 8a58c06adb3604bcd59ce2ec5c78b58e71f0cdc5  (2016) vs b8f9f171e06c273e0fdd5d5d19eddcc0744fd26f (2020) for IGNORE_RETURN

**Review by tridge — APPROVED** [review]

**Comment by peterbarker** [comment]
@muramura please rebase to get CI run again

**Comment by hendjoshsr71** [comment]
@muramura san I've rebased this for you.

**Pull request merged**